tests/datatype/tests_time_types.cpp

```diff
@@ -554,3 +554,55 @@ TEST_CASE("Literal API") {
     CHECK(Literal::make_typed<datatypes::xsd::GDay>("3").as_tz() == Literal::make_simple(""));
     CHECK(Literal::make_simple("5").as_tz() == Literal{});
 }
+
+TEST_CASE("arithmetic") {
```

Comment:
Todo to self: look at spec and check tests

Reply:
If I remember correctly you based the behaviour on some reference implementations, correct? If so please mention somewhere what the current behaviour was inspired by.